### PR TITLE
Add PoseArray display plugin

### PR DESCRIPTION
**Pull request by Sarath18**
Render `geometry_msgs/msg/PoseArray` data received on ROS2 topics.

The sample test script can be found [here](https://gist.github.com/Sarath18/c25ebd015ca4bf4793150f94387c7a2f#file-posearraypublisher-py).

### CODE REVIEW

File: ign_rviz_plugins/include/ignition/rviz/plugins/PoseArrayDisplay.hpp [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+#include "ignition/rviz/plugins/message_display_base.hpp"
+
+namespace ignition
+{
+namespace rviz
+{
+namespace plugins
+{
+struct PoseArrayVisual
```
Comment (ahcorde) [inline ign_rviz_plugins/include/ignition/rviz/plugins/PoseArrayDisplay.hpp:37]: Why is this part of a new struct? Can you define these variables and method inside the class ?

Comment (Sarath18) [reply ign_rviz_plugins/include/ignition/rviz/plugins/PoseArrayDisplay.hpp:37]: The idea was to abstract to the PoseArray visuals. Going through the code again, it looks like I am not using the abstraction as properly. I can move it to the main class.

File: ign_rviz_plugins/include/ignition/rviz/plugins/PoseArrayDisplay.hpp [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  void onRefresh();
+
+protected:
+  /**
+   * @brief Update PoseArray data visualization
+   */
+  void update() override;
+
+  /**
+   * @brief Update pose array visual dimentions
```
Comment (ahcorde) [inline ign_rviz_plugins/include/ignition/rviz/plugins/PoseArrayDisplay.hpp:173]:
```suggestion
   * @brief Update pose array visual dimensions
```

File: ign_rviz_plugins/src/rviz/plugins/PoseArrayDisplay.cpp [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+    update();
+  }
+
+  return QObject::eventFilter(_object, _event);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void PoseArrayDisplay::reset()
+{
+//   this->arrow.visual->SetLocalPose(math::Pose3d::Zero);
```
Comment (ahcorde) [inline ign_rviz_plugins/src/rviz/plugins/PoseArrayDisplay.cpp:125]: Remove these lines or add a TODO

Comment (Sarath18) [reply ign_rviz_plugins/src/rviz/plugins/PoseArrayDisplay.cpp]: I implemented the reset for visuals in aff5843

File: ign_rviz_plugins/include/ignition/rviz/plugins/PoseArrayDisplay.hpp (hunk 2 above)
Comment (Sarath18) [reply ign_rviz_plugins/include/ignition/rviz/plugins/PoseArrayDisplay.hpp]: My bad. I even changed the spelling function names. PoseDisplay has this same issue, should I fix it in this PR itself?

**Pull request merged**